libcxx/test/std/numerics/c.math/hermite.pass.cpp

```diff
@@ -0,0 +1,310 @@
+            T(3.17699916197996),
+            T(3.86944790486012),
+            T(4.68873893930582)};
+
+  case 17:
+    return {T(0),
+            T(0.5316330013427),
+            T(1.0676487257435),
+            T(1.6129243142212),
+            T(2.1735028266666),
+            T(2.7577629157039),
+            T(3.3789320911415),
+            T(4.0619466758755),
+            T(4.8713451936744)};
+
+  case 18:
+    return {T(0.2582677505191),
+            T(0.7766829192674),
+            T(1.3009208583896),
+            T(1.8355316042616),
+            T(2.3862990891667),
+            T(2.9613775055316),
+            T(3.5737690684863),
+            T(4.2481178735681),
+            T(5.0483640088745)};
+
+  case 19:
+    return {T(0),
+            T(0.5035201634239),
+            T(1.0103683871343),
+            T(1.5241706193935),
+            T(2.0492317098506),
+            T(2.5911337897945),
+            T(3.1578488183476),
+            T(3.7621873519640),
+            T(4.4285328066038),
+            T(5.2202716905375)};
+
+  case 20:
+    return {T(0.2453407083009),
+            T(0.7374737285454),
+            T(1.2340762153953),
+            T(1.7385377121166),
+            T(2.2549740020893),
+            T(2.7888060584281),
+            T(3.347854567332),
+            T(3.9447640401156),
+            T(4.6036824495507),
+            T(5.3874808900112)};
+
+  default:
+    throw "Roots of Hermite polynomial of order " + std::to_string(n) + " not implemented!\n";
+  }
+}
+
+/// \param [in] Tolerance of the root. This value must be smaller than
+/// the smallest difference between adjacent roots in the given range
+/// with n <= 20.
+template <class T>
+void test_roots(T Tolerance) {
+  const auto is_sign_change = [Tolerance](unsigned n, T x) -> bool {
+    return std::hermite(n, x - Tolerance) * std::hermite(n, x + Tolerance) < 0;
+  };
+
+  for (unsigned n = 0; n <= 20u; ++n) {
+    for (T x : get_roots<T>(n)) {
+      // the roots are symmetric: if x is a root, so is -x
+      if (x > 0)
+        assert(is_sign_change(n, -x));
+      assert(is_sign_change(n, x));
+    }
+  }
+}
+
+template <class T>
+void test_hermite(T abs_tol, T rel_tol) {
+  test_NaN_propagation<T>();
+
+  for (T x : sample_points<T>()) {
+    test_not_NaN(x);
+    test_analytic_solution(x, abs_tol, rel_tol);
+  }
+}
+
+template <class Integer>
+void test_integers() {
+  for (unsigned n = 0; n < MAX_N; ++n)
+    for (Integer x : {-1, 0, 1})
+      assert(std::hermite(n, x) == std::hermite(n, static_cast<double>(x)));
+}
+
+void test_hermitef() {
+  for (unsigned n = 0; n < MAX_N; ++n)
+    for (float x : sample_points<float>())
+      assert(std::hermite(n, x) == std::hermitef(n, x));
+}
+
+void test_hermitel() {
+  for (unsigned n = 0; n < MAX_N; ++n)
+    for (long double x : sample_points<long double>())
+      assert(std::hermite(n, x) == std::hermitel(n, x));
+}
+} // namespace
+
+int main(int, char**) {
+  test_hermite<float>(1e-5f, 1e-5f);
```

Comment:
Please refactor the tests to the style
```c++
template <class T>
void test() {
  { // Write what the test checks
    // put the test here
  }

  { // and the next test
    // some code
  }
}
```
We generally use this in new tests.

Reply:
Do I understand you correctly that the `main` should only call 3 `test<T>();` functions (for `T=float, double, long double`)?

Reply:
Implemented in 8b57a634c764c2aef653106fbb6b762663f9cf75.